connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -283,49 +296,140 @@ public void testReplication() throws InterruptedException {
-            backup.kafka().produce("test-topic-3", 0, "key", "message-3-" + i);
+        produceMessages(primary, "test-topic-2", "message-3-", 1);
+        produceMessages(backup, "test-topic-3", "message-4-", 1);
+
+        assertEquals("Records were not produced to primary cluster.", NUM_RECORDS_PER_PARTITION,
+            primary.kafka().consume(NUM_RECORDS_PER_PARTITION, RECORD_TRANSFER_DURATION_MS, "test-topic-2").count());
+        assertEquals("Records were not produced to backup cluster.", NUM_RECORDS_PER_PARTITION,
+            backup.kafka().consume(NUM_RECORDS_PER_PARTITION, RECORD_TRANSFER_DURATION_MS, "test-topic-3").count());
+
+        assertEquals("New topic was not replicated to primary cluster.", NUM_RECORDS_PER_PARTITION,
+            primary.kafka().consume(NUM_RECORDS_PER_PARTITION, 2 * RECORD_TRANSFER_DURATION_MS, "backup.test-topic-3").count());
+        assertEquals("New topic was not replicated to backup cluster.", NUM_RECORDS_PER_PARTITION,
+            backup.kafka().consume(NUM_RECORDS_PER_PARTITION, 2 * RECORD_TRANSFER_DURATION_MS, "primary.test-topic-2").count());
+    }
+
+    @Test
+    public void testReplicationWithEmptyPartition() throws InterruptedException {
+        String consumerGroupName = "consumer-group-testReplicationWithEmptyPartition";
+        Map<String, Object> consumerProps  = new HashMap<String, Object>() {{
+                put("group.id", consumerGroupName);
+                put("auto.offset.reset", "latest");
+            }};
+
+        // create topics
+        String topic = "test-topic-empty";
+        String primaryTopicReplica = "primary." + topic;
+        String backupTopicReplica = "backup." + topic;
+        primary.kafka().createTopic(topic, NUM_PARTITIONS);
+        primary.kafka().createTopic(backupTopicReplica, 1);
+        backup.kafka().createTopic(topic, NUM_PARTITIONS);
+        backup.kafka().createTopic(primaryTopicReplica, 1);
+
+        // Consume, from the primary cluster, before starting the connectors so we don't need to wait for discovery
+        Consumer<byte[], byte[]> consumer = primary.kafka().createConsumerAndSubscribeTo(consumerProps, topic, backupTopicReplica);
+        consumeAllMessages(consumer, 0);
+
+        // produce to all test-topic-empty's partitions *but the last one*, on the primary cluster
+        produceMessages(primary, topic, "message-1-", NUM_PARTITIONS - 1);
+
+        // Consume all messages
+        consumeAllMessages(consumer, NUM_RECORDS_PER_PARTITION * (NUM_PARTITIONS - 1));
+        consumer.close();
+
+        // Consumer group offsets after consumption: topic's last partition doesn't yet has data, so
+        // the committed offset is 0. All other topic's partition should have offset equal to NUM_RECORDS_PER_PARTITION.
+        // backupTopicReplica still has a single empty partition, since MM2 is not yet started, and its record offset is 0.
+
+        mm2Config = new MirrorMakerConfig(mm2Props);
```

Comment:
this overwrite of mm2config should go in the setup method, IMHO

Reply:
Thanks, @edoardocomar . I've addressed this.